### ISSUE TITLE
Exception Not Handled in RELEASE builds under .NET 8 MAUI (Android)

### ISSUE THREAD

**Issue by DAConsulting** [dotnet/maui#24545]
### Description

Exceptions generated by the system are not "caught" in a release build of an Android MAUI application, IF a background thread has run that uses System.Net.NetworkInformation.Ping.  Instead, the catch handler is never called and the application abends and quits. Note that it is not necessary for the background thread to be running to cause this issue when the exception occurs.  If the background thread was run once, then stopped, the problem will still occur.

### Steps to Reproduce

1.  Download the ZIP file containing the demo app from the link below.
2.  Unzip and build a RELEASE version of the app on an Android device.
3.  Press the button to force an exception a few times.  This will generate a null reference exception.
4.  You should see an alert displayed from the catch() block of the exception handler in the code.
5.  Now press the button to start the Background Ping thread.
6.  Press the button to force the exception again, after having started the background thread
7.  This time, the catch() handler will never execute and the app will just abend.

### Link to public reproduction project repository

https://drive.google.com/file/d/1mCSkiVGwSh3aHpwCZXhUb-AvEo4QNttv/view?usp=sharing

### Version with bug

8.0.3 GA

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 31 Targeted

### Did you find any workaround?

Only workaround is to NOT use the Ping class in the background thread.

Bjuild logs are attached here, since there  is no way to upload from "Relevant log output" below.

[TestException_Release_Build_Logs.zip](https://github.com/user-attachments/files/16820305/TestException_Release_Build_Logs.zip)

**Comment by ninachen03** [comment]
This issue has been verified using latest Visual Studio 17.12.0 Preview 1.0 (8.0.82) & 17.11.2 NET7 7.0.101, Can repro on Android platform

**Comment by drasticactions** [comment]
I don't believe this will be a MAUI UI issue, it seems like a [runtime](https://github.com/dotnet/runtime) bug, @jonathanpeppers What do you think?

**Comment by jonathanpeppers** [comment]
Yes, `System.Net.NetworkInformation.Ping`, so moving to dotnet/runtime.
